File: backend/app/schemas/elliott_wave.py
from pydantic import BaseModel

from app.schemas.signals import RiskContext


class WavePivot(BaseModel):
    index: int
    date: str
    price: float
    type: str
    wave_label: str


class FibLevel(BaseModel):
    level: float
    ratio: str
    label: str


class WaveAnalysis(BaseModel):
    pattern: str
    current_wave: str
    confidence: float
    wave_pivots: list[WavePivot]
    fib_levels: list[FibLevel]


class IndividualSignal(BaseModel):
    name: str
    score: float
    description: str


class ElliottWaveResult(BaseModel):
    symbol: str
    current_price: float
    wave_analysis: WaveAnalysis
    signals: list[IndividualSignal]
    direction: str
    conviction: str
    risk: RiskContext

File: frontend/src/types/elliott_wave.ts
export interface WavePivot {
  index: number;
  date: string;
  price: number;
  type: "high" | "low";
  wave_label: string;
}

export interface FibLevel {
  level: number;
  ratio: string;
  label: string;
}

export interface WaveAnalysis {
  pattern: string;
  current_wave: string;
  confidence: number;
  wave_pivots: WavePivot[];
  fib_levels: FibLevel[];
}

export interface IndividualSignal {
  name: string;
  score: number;
  description: string;
}

export interface RiskContext {
  stop_loss: number | null;
  target_price: number | null;
  risk_reward: number | null;
  position_size: number;
  position_pct: number;
}

export interface ElliottWaveResult {
  symbol: string;
  current_price: number;
  wave_analysis: WaveAnalysis;
  signals: IndividualSignal[];
  direction: "buy" | "sell" | "hold";
  conviction: "low" | "medium" | "high";
  risk: RiskContext;
}
